Zend/zend_compile.c

```diff
@@ -7375,6 +7521,34 @@ void zend_compile_class_decl(znode *result, zend_ast *ast, zend_bool toplevel) /
 		zend_compile_implements(implements_ast);
 	}
 
+	if (ce->ce_flags & ZEND_ACC_ENUM) {
+		zend_string *cases_func_name = zend_string_init("cases", strlen("cases"), 1);
+		zend_internal_function *cases_function = malloc(sizeof(zend_internal_function));
+		memset(cases_function, 0, sizeof(zend_internal_function));
+		cases_function->type = ZEND_INTERNAL_FUNCTION;
+		cases_function->module = EG(current_module);
+		cases_function->handler = zend_enum_cases_func;
+		cases_function->function_name = cases_func_name;
+		cases_function->scope = ce;
+		cases_function->fn_flags = ZEND_ACC_PUBLIC|ZEND_ACC_STATIC;
```

Comment:
Should this be final (`|ZEND_ACC_FINAL`) here and below?

Reply:
The whole class is final so this is unnecessary, no?

Reply:
Oh, right, forgot that